batch_seqr_loader/batch_workflow.py

```diff
@@ -437,6 +455,119 @@ def _pedigree_checks(
     return check_j, somalier_samples_path
 
 
+def _make_realign_bam_jobs(
+    b: hb.Batch,
+    samples_df: pd.DataFrame,
+    file_type: str,  # 'fastq_to_realign', 'cram_to_realign'
+    reference: hb.ResourceGroup,
+    work_bucket: str,
+    overwrite: bool,
+) -> List[Job]:
+    """
+    Takes all samples with a 'file' of 'type'='fastq_to_realign'|'cram_to_realign'
+    in `samples_df`, runs BWA to realign reads again, and sets a new 'file'
+    of 'type'='cram'.
+
+    When the input is CRAM/BAM, uses Bazam to stream reads to BWA.
+    """
+    jobs = []
+    realign_df = samples_df[samples_df['type'] == file_type]
+    for sn, file1, file2, index in zip(
+        realign_df['s'], realign_df['file'], realign_df['file2'], realign_df['index']
+    ):
+        job_name = f'BWA align, {sn}'
+        output_cram_path = join(work_bucket, f'{sn}.cram')
+        if can_reuse(output_cram_path, overwrite):
+            jobs.append(b.new_job(f'{job_name} [reuse]'))
+        else:
+            assert (
+                (file1.endswith('.cram') or file1.endswith('.bam')) and index or file2
+            )
+            if file1.endswith('.cram') or file1.endswith('.bam'):
+                use_bazam = True
+                file1 = b.read_input_group(base=file1, index=index)
+            else:
+                use_bazam = False
+                file1 = b.read_input(file1)
+                file2 = b.read_input(file2)
+
+            j = b.new_job(job_name)
+            jobs.append(j)
+            j.image(BAZAM_CONTAINER)
+            total_cpu = 16
+            if use_bazam:
+                bazam_cpu = 3
+                bwa_cpu = 10
+                bamsormadup_cpu = 3
+            else:
+                bazam_cpu = 0
+                bwa_cpu = 12
+                bamsormadup_cpu = 4
+            j.memory('highmem')
```

Comment:
For the `n1` generation, that would give you 104 GB of RAM for 16 cores total, I think. Just checking that 60 GB (for `standard` instead of `highmem`) wouldn't be enough?

Reply:
I updated the `standard` machine to use 32 cores, it works well now! Though will explore creating a custom pool later.